packages/contracts-gen/src/contracts-gen.ts

```diff
@@ -0,0 +1,171 @@
+
+(async () => {
+    const packageDir = process.cwd();
+    const compilerJSON = readJSONFile<CompilerOptions>('compiler.json');
+    const contracts = compilerJSON.contracts;
+    const contractsDir = compilerJSON.contractsDir || DEFAULT_CONTRACTS_DIR;
+    const artifactsDir = compilerJSON.artifactsDir || DEFAULT_ARTIFACTS_DIR;
+    const wrappersDir = DEFAULT_WRAPPERS_DIR;
+    if (!_.isArray(contracts)) {
+        throw new Error('Unable to run the generator bacause contracts key in compiler.json is not of type array');
+    }
+    const prettierConfig = await prettier.resolveConfig(packageDir);
+    generateCompilerJSONContractsList(contracts, contractsDir, prettierConfig);
+    generateArtifactsTs(contracts, artifactsDir, prettierConfig);
+    generateWrappersTs(contracts, wrappersDir, prettierConfig);
+    generateTsConfigJSONFilesList(contracts, artifactsDir, prettierConfig);
+    generatePackageJSONABIConfig(contracts, artifactsDir, prettierConfig);
+    process.exit(0);
+})().catch(err => {
+    logUtils.log(err);
+    process.exit(1);
+});
+
+function generateCompilerJSONContractsList(
+    contracts: string[],
+    contractsDir: string,
+    prettierConfig: prettier.Options | null,
+): void {
+    const COMPILER_JSON_FILE_PATH = 'compiler.json';
+    const compilerJSON = readJSONFile<CompilerOptions>(COMPILER_JSON_FILE_PATH);
+    compilerJSON.contracts = _.map(contracts, contract => {
+        if (contract.endsWith(SOLIDITY_EXTENSION)) {
+            // If it's already a relative path - NO-OP.
+            return contract;
+        } else {
+            // If it's just a contract name - resolve it and rewrite.
+            return new NameResolver(contractsDir).resolve(contract).path;
+        }
+    });
+    compilerJSON.contracts = _.sortBy(compilerJSON.contracts);
+    const compilerJSONString = JSON.stringify(compilerJSON);
+    const formattedCompilerJSON = prettier.format(compilerJSONString, {
+        ...prettierConfig,
+        filepath: COMPILER_JSON_FILE_PATH,
+    });
+    fs.writeFileSync(COMPILER_JSON_FILE_PATH, formattedCompilerJSON);
+}
+
+function generateArtifactsTs(contracts: string[], artifactsDir: string, prettierConfig: prettier.Options | null): void {
+    const imports = _.map(contracts, contract => {
+        const contractName = path.basename(contract, SOLIDITY_EXTENSION);
+        const importPath = path.join('..', artifactsDir, `${contractName}.json`);
+        return `import * as ${contractName} from '${importPath}';`;
+    });
+    const sortedImports = _.sortBy(imports);
+    const artifacts = _.map(contracts, contract => {
+        const contractName = path.basename(contract, SOLIDITY_EXTENSION);
+        if (contractName === 'ZRXToken') {
+            // HACK(albrow): "as any" hack still required here because ZRXToken does not
+            // conform to the v2 artifact type.
+            return `${contractName}: (${contractName} as any) as ContractArtifact,`;
```

Comment:
Is this HACK still required? My understanding is we are regenerating all artifacts so the ZRXToken would conform to the new artifact format.

Reply:
We're using the old version of Solidity and it still does not confirm.

Reply:
ABI types changed between solidity versions